apps/web/pages/settings/my-account/profile.tsx

```diff
@@ -76,31 +77,72 @@ type FormValues = {
 };
 
 const ProfileView = () => {
-  const { t } = useLocale();
+  const { t, i18n } = useLocale();
   const utils = trpc.useContext();
   const { data: user, isLoading } = trpc.viewer.me.useQuery();
   const { data: avatar, isLoading: isLoadingAvatar } = trpc.viewer.avatar.useQuery();
-  const mutation = trpc.viewer.updateProfile.useMutation({
-    onSuccess: () => {
+  const updateProfileMutation = trpc.viewer.updateProfile.useMutation({
+    onSuccess: async (res) => {
       showToast(t("settings_updated_successfully"), "success");
+      if (res.googleEmailChange && tempFormValues) {
+        setLoading(true);
```

Comment:
you can use updateProfileMutation.loading for this kind of state
```suggestion
```

Reply:
Does the isLoading cover the code run under onSuccess too? I was using a separate state since I was under the assumption that isLoading becomes false as soon as the mutation returns. I couldn't find anything on the trpc docs. I will try and will let you know.

Reply:
When it's onSuccess it's not really no, I just reread the logic and now I'm wondering if the password reset request should be triggered in the backend; instead of a follow up on user profile update.

Reply:
Yup the password reset logic is being triggered. But since there is no password before as the user would have been using Google for authentication, we don't need to actually trigger the logic. I am just sending the forgot password email through which user can set a new password.